otter/integration/tests/test_convergence.py

```diff
@@ -0,0 +1,212 @@
+    BreakLoopException,
+    ScalingGroup,
+    ScalingPolicy,
+)
+from otter.integration.lib.identity import IdentityV2
+from otter.integration.lib.resources import TestResources
+
+from otter.util.deferredutils import retry_and_timeout
+from otter.util.http import check_success, headers
+from otter.util.retry import (
+    TransientRetryError,
+    repeating_interval,
+    transient_errors_except,
+)
+
+
+username = os.environ['AS_USERNAME']
+password = os.environ['AS_PASSWORD']
+endpoint = os.environ['AS_IDENTITY']
+flavor_ref = os.environ['AS_FLAVOR_REF']
+image_ref = os.environ['AS_IMAGE_REF']
+region = os.environ['AS_REGION']
+
+
+def find_end_points(rcs):
+    """Locates the endpoints we need to conduct convergence tests."""
+
+    rcs.token = rcs.access["access"]["token"]["id"]
+    sc = rcs.access["access"]["serviceCatalog"]
+    try:
+        rcs.endpoints["otter"] = auth.public_endpoint_url(sc,
+                                                          "autoscale",
+                                                          region)
+    except auth.NoSuchEndpoint:
+        # If the autoscale endpoint is not defined, use local otter
+        rcs.endpoints["otter"] = 'http://localhost:9000/v1.0/{0}'.format(
+            rcs.access['access']['token']['tenant']['id'])
+
+    rcs.endpoints["loadbalancers"] = auth.public_endpoint_url(
+        sc, "cloudLoadBalancers", region
+    )
+    rcs.endpoints["nova"] = auth.public_endpoint_url(
+        sc, "cloudServersOpenStack", region
+    )
+    return rcs
+
+
+class TestConvergence(unittest.TestCase):
+    """This class contains test cases aimed at the Otter Converger."""
+
+    def setUp(self):
+        """Establish an HTTP connection pool and commonly used resources for
+        each test.  The HTTP connection pool is important for maintaining a
+        clean Twisted reactor.
+        """
+
+        self.pool = HTTPConnectionPool(reactor, False)
+        self.identity = IdentityV2(
+            auth=auth, username=username, password=password,
+            endpoint=endpoint, pool=self.pool
+        )
+
+    def tearDown(self):
+        """Destroy the HTTP connection pool, so that we close the reactor
+        cleanly.
+        """
+
+        def _check_fds(_):
+            fds = set(reactor.getReaders() + reactor.getReaders())
+            if not [fd for fd in fds if isinstance(fd, Client)]:
+                return
+            return deferLater(reactor, 0, _check_fds, None)
+        return self.pool.closeCachedConnections().addBoth(_check_fds)
+
+    def test_reaction_to_oob_server_deletion(self):
+        """Exercise out-of-band server deletion.  The goal is to spin up, say,
+        eight servers, then use Nova to delete four of them.  We should be able
+        to see, over time, more servers coming into existence to replace those
+        deleted.
+        """
+
+        N_SERVERS = 4
+
+        rcs = TestResources()
+
+        scaling_group_body = {
```

Comment:
For now, I would just suggest a function `otter.integration.lib.autoscale.scaling_group_body_template()` that returns the `dict`. Ideally, it would be nice to have `create_scaling_group(img_ref=None, flavor_ref=None)` function that defaults to configured (via env or config file) image ref, flavor ref and other group properties if it is not provided.